packages/editor/src/components/block-list/block-invalid-warning.js

```diff
@@ -96,18 +97,23 @@ const blockToBlocks = ( block ) => rawHandler( {
 	HTML: block.originalContent,
 } );
 
-export default withDispatch( ( dispatch, { block } ) => {
-	const { replaceBlock } = dispatch( 'core/editor' );
+export default compose( [
+	withSelect( ( select, { clientId } ) => ( {
+		block: select( 'core/editor' ).getBlock( clientId ),
```

Comment:
This looks like a good candidate for optimization when #11851 lands.

**Edit:** I guess it probably won't have much an impact in common usage, if the block warning is not expected to be frequently shown. I don't see any downside to standardizing on a convention of using the `registry` argument if the selected prop is only used in the callback of a `withDispatch` function though.